src/main/flight/alt_hold.c

```diff
@@ -0,0 +1,241 @@
+        // iTerm is relatively weak, intended to be slow moving to adjust baseline errors
+        // the Hover value is important otherwise takes time for iTerm to correct
+        // High P will wobble readily
+        // with these scalers, the same numbers as for GPS Rescue work OK for altHold
+        // the goal is to share these gain factors, if practical for all altitude controllers
+
+    //setup altitude D filter
+    const float cutoffHz = 0.01f * positionConfig()->altitude_d_lpf; // default 1Hz, time constant about 160ms
+    const float gain = pt2FilterGain(cutoffHz, taskIntervalSeconds);
+    pt2FilterInit(&altHoldDeltaLpf, gain); 
+
+    altHoldState.isAltHoldActive = false;
+    altHoldReset();
+}
+
+void altHoldProcessTransitions(void) {
+
+    if (FLIGHT_MODE(ALTHOLD_MODE)) {
+        if (!altHoldState.isAltHoldActive && isAltitudeAvailable()) {
+            altHoldReset();
+            altHoldState.isAltHoldActive = true;
+        }
+    } else {
+        altHoldState.isAltHoldActive = false;
+    }
+
+    // ** the transition out of alt hold (exiting altHold) may be rough.  Some notes... **
+    // The original PR had a gradual transition from hold throttle to pilot control throttle
+    // using !(altHoldRequested && altHoldState.isAltHoldActive) to run an exit function
+    // a cross-fade factor was sent to mixer.c based on time since the flight mode request was terminated
+    // it was removed primarily to simplify this PR
+
+    // hence in this PR's the user's throttle needs to be close to the hover throttle value on exiting altHold
+    // its not so bad because the 'target adjustment' by throttle requires that
+    // user throttle must be not more than half way out from hover for a stable hold
+}
+
+void altHoldUpdateTargetAltitude(void)
+{
+    // The user van raise or lower the target altitude with throttle up;  there is a big deadband.
+    // Max rate for climb/descend is 1m/s by default (up to 2.5 is allowed but overshoots a fair bit)
+    // If set to zero, the throttle has no effect.
+
+    // Some people may not like throttle being able to change the target altitude, because:
+    // - with throttle adjustment, hitting the switch won't always hold altitude if throttle is bumped
+    // - eg if the throttle is bumped low accidentally, quad will start descending.
+    // On the plus side:
+    // - the pilot has control nice control over altitude, and the deadband is wide
+    // - Slow controlled descents are possible, eg for landing
+    // - fine-tuning height is possible, eg if there is slow sensor drift
+    // - to keep the craft stable, throttle must be in the deadband, making exits smoother
+
+    const float rcThrottle = rcCommand[THROTTLE];
+
+    const float lowThreshold = 0.5f * (positionConfig()->hover_throttle + PWM_RANGE_MIN); // around 1150
+    const float highThreshold = 0.5f * (positionConfig()->hover_throttle + PWM_RANGE_MAX); // around 1500
```

Comment:
No, but my original reasoning was that I thought that a fairly large dead-band would make it easy for the pilot to get the stick into a point where the target was fixed, and I wanted it centered around hover, to stabilise the exit from the mode.  For high authority quads the low end would always be 'tight' and the high end 'loose'.  On defaults, a quad with hover at 1300 would have a range -150 to +350, which is quite wide.  A quad with hover at 1400, a somewhat under-powered build, would be more symmetrical at -200 to +300.  

I'll experiment with making a symmetrical threshold either side of hover and see how easily I can get my stick into the 'null' range.

It would be nice to be symmetrical as it would simplify the code.

We can see what other testers think, and modify this in future quite easily based on user feedback.  At present it is 'safe' but I have to admit I would probably prefer the tighter range, and I'll test it out.  Thanks prompting me to do that.